src/core/lender/distributor/switchable.sol

```diff
@@ -39,15 +35,9 @@ contract SwitchableDistributor is Distributor {
         }  else revert();
     }
 
-    function depend(bytes32 what, address addr) public auth {
-        if (what == "currency") {
-            currency = CurrencyLike(currency);
-        }  else revert();
-    }
-
     function balance() public {
         if(borrowFromTranches) {
-            uint currencyAmount = add(senior.balance(), junior.balance());
+            uint currencyAmount = add(currency.balanceOf(address(senior)), currency.balanceOf(address(junior)));
```

Comment:
ahh didn't see, that one. I would prefer if the tranche can tell how much currency is available to take.

It allows more flexibility in the tranche impl.